### PR TITLE
[CLI-324] Make adding OptionGroups and Options to existing Options easier

### PR DESCRIPTION
**Pull request by Claudenw**
Fix for CLI-324 https://issues.apache.org/jira/browse/CLI-324

### CODE REVIEW

**Comment by garydgregory** [comment]
@Claudenw 
Thank you for your PR.
You must provide tests for new features and updates.

File: src/main/java/org/apache/commons/cli/Options.java [hunk 1]
```diff
@@ -231,8 +243,9 @@ public OptionGroup getOptionGroup(final Option opt) {
      * Gets the OptionGroups that are members of this Options instance.
      *
      * @return a Collection of OptionGroup instances.
+     * @since 1.7.0
      */
-    Collection<OptionGroup> getOptionGroups() {
+    public Collection<OptionGroup> getOptionGroups() {
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/cli/Options.java:248]: Hello @Claudenw 
The typing needs some thought I think if make this public: 
- First, do we really want this to be public?
- Why are we filtering out duplicates here? 
- If we indeed want to return a Set, why hide this and type the return as a Collection.
Consider:
- Changing the return type to a Set
- Changing the wrapping to an ArrayList

Thoughts?

**Review by garydgregory — CHANGES_REQUESTED** [review]

File: src/main/java/org/apache/commons/cli/Options.java (hunk 1 above)
Comment (Claudenw) [reply src/main/java/org/apache/commons/cli/Options.java:248]: Reverting the change.  I put it in in my codebase when the Options#addOptions(Options) method did not exist so I could implement that.  The code is what was originally in the code base prior to 1.7.0

I'll add some notes to the code base to answer your questions, but basically. the OptionGroups is a map of Option to OptionGroup.  So the OptionGroup (value) will occur multiple times in the Map#getValues() list.  Creating a Set simply filters them out.  Returning it as a collection seems appropriate as Options are returned as a collection and it keeps a symmetry.

File: src/main/java/org/apache/commons/cli/Options.java [hunk 2]
```diff
@@ -54,6 +54,18 @@ public class Options implements Serializable {
     /** A map of the option groups */
     private final Map<String, OptionGroup> optionGroups = new LinkedHashMap<>();
 
+    /**
+     * Adds options to this option.
+     * @param options the options to add.
+     * @return this for chaining.
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/cli/Options.java:60]: We do not need to state the purpose of returning `this` as fluent style coding since the return value can be used for _anything_.

File: src/test/java/org/apache/commons/cli/OptionsTest.java [hunk 3]
```diff
@@ -78,6 +79,26 @@ public void testGetOptionsGroups() {
...
+        group1.addOption(Option.builder("a").build());
+        group1.addOption(Option.builder("b").build());
+
+        options.addOptionGroup(group1);
+
+        options.addOption(Option.builder("X").build());
+        options.addOption(Option.builder("y").build());
+
+        final Options underTest = new Options();
+        underTest.addOptions(options);
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/cli/OptionsTest.java:96]: You need a test that calls `addOptions()` twice with different options to verify that the calls are cumulative and do not replace what's already there. 

Also: What should happen if I add the same options twice? For example:
```
underTest.addOptions(options);
underTest.addOptions(options);
```
Also test adding an empty Options object to make sure nothing odd happens. 

Edge cases, edge cases... ;-)

**Review by garydgregory — CHANGES_REQUESTED** [review]
@Claudenw 
TY for your updates. Please see my comments.

File: src/test/java/org/apache/commons/cli/OptionsTest.java (hunk 3 above)
Comment (Claudenw) [reply src/test/java/org/apache/commons/cli/OptionsTest.java:101]: Updated.  I think I got them nailed down.
